src/jit/compiler.cpp

```diff
@@ -4191,6 +4191,16 @@ void Compiler::compCompile(void** methodCodePtr, ULONG* methodCodeSize, JitFlags
 
     fgImport();
 
+    if (IsTargetAbi(CORINFO_CORERT_ABI) && hasFatPointer())
+    {
+        if (fgCheapPredsValid)
+        {
+            fgRemovePreds();
+        }
+        fgTransformFatCalli();
+        setHasFatPointer(false);
+    }
+
     assert(!fgComputePredsDone);
```

Comment:
Couldn't you move this code below the `if (fgCheapPredsValid)` case, below, and then remove the `if (fgCheapPredsValid)` code in this `if`? Also, could you put the `setHasFatPointer(false)` at the end of `fgTransformFatCalli()`? In this function, I'd prefer that as much as possible, it's just a linear listing of phases.

Also, is this going to take long enough (e.g., do an IR pass / tree walk) such that you should add a phase name for it, and thus an `EndPhase()` call?